packages/cli/src/commands/deploy/index.ts

```diff
@@ -512,18 +512,24 @@ export default async (client: Client): Promise<number> => {
 
   const localConfigurationOverrides = pickOverrides(localConfig);
 
+  const name = project ? project.name : newProjectName;
+  if (!name) {
+    throw new Error(
+      '`name` not found on project or provided by existing project'
+    );
+  }
+
   try {
-    const createArgs: any = {
-      name: project ? project.name : newProjectName,
+    const createArgs: CreateOptions = {
+      name,
       env: deploymentEnv,
       build: { env: deploymentBuildEnv },
       forceNew: argv['--force'],
       withCache: argv['--with-cache'],
       prebuilt: argv['--prebuilt'],
       rootDirectory,
       quiet,
-      wantsPublic: argv['--public'] || localConfig.public,
-      type: null,
+      wantsPublic: (argv['--public'] || localConfig.public) ?? false,
```

Comment:
```suggestion
      wantsPublic: Boolean(argv['--public'] || localConfig.public),
```

Nit pick: this reads better to me since its less logic and more about the type

Reply:
I thought our style guide prohibited `Boolean()`, so I've been doing `!!`.

Reply:
It looks like the style guide prefers `Boolean()`: https://github.com/vercel/style-guide/blob/main/eslint/rules/best-practice.js#L93-L98

Reply:
Ha!